src/graphology/src/graph.js

```diff
@@ -408,12 +408,7 @@ function mergeEdge(
     if (edgeData) {
       // Here, we need to ensure, if the user gave a key, that source & target
       // are coherent
-      if (
-        edgeData.source.key !== source ||
-        edgeData.target.key !== target ||
-        (undirected &&
-          (edgeData.source.key !== target || edgeData.target.key !== source))
-      ) {
+      if (!(edgeData.source.key == source && edgeData.target.key == target || undirected && (edgeData.source.key == target && edgeData.target.key == source))) {
```

Comment:
1. You should rely on `===`, not on `==`.
2. This condition is less peformant because of the `&&` since it cannot early exit. I think it might be time for a ternary or a more explicit condition handling the undirected case, what do you think?

Reply:
You're right - I think a second conditional for the undirected case is appropriate. I've committed a rewrite - let me know what you think of it.